pkg/operator/context/apis.go

```diff
@@ -59,9 +85,37 @@ func getAPIs(config *userconfig.Config,
 					ResourceType: resource.APIType,
 				},
 			},
-			API:  apiConfig,
-			Path: context.APIPath(apiConfig.Name, config.App.Name),
+			API:                   apiConfig,
+			Path:                  context.APIPath(apiConfig.Name, config.App.Name),
+			RequestHandlerImplKey: requestHandlerImplKey,
+		}
+
+		if apiConfig.Requesthandler != nil {
+			uploadRequestHandlers(apis[apiConfig.Name], impls[*apiConfig.Requesthandler])
 		}
 	}
 	return apis, nil
 }
+
+func uploadRequestHandlers(api *context.API, impl []byte) error {
```

Comment:
If you move the call to this within the first `if apiConfig.Requesthandler != nil {` above, then you can also pass in `implID` to save the double hashing and separate the concerns a little better too, does that work?